### DISCUSSION TITLE
User Tracking

### DISCUSSION THREAD

**Discussion by sksdev27** [open-webui/open-webui#6605] [General]
I am researching OpenWebUI and deploying it in an environment where it is being used by a lot of people. We are researching scaling this to a even bigger audience and one of the metrics we want to establish is how many people can cause the GPU or the application to break. I couldn't find any pipeline associated with it. I see that it does show which users are active and last were active but there's not like how many users were active and how much usage did they do. Any help on this would be greatly appreciated. I apologize in advance if this feature is implemented in pipeline or within OpenWebUI planning. Please let me know if it is. Thank you again.

**Comment by ladbyte** [comment]
I also hope to increase the statistics of user usage, and it is best to have an additional information of hardware resources for easy analysis. Thanks again for this excellent project.

**Comment by MindSetFPS** [comment]
This would be a useful feature. In my case I would appreciate global token input and outputs counters, as well as per user.

**Reply by VariantConst** [reply]
I just developed a monitor for OpenWebUI that tracks user activity, allows setting quotas for users, and notifies them of the token usage and cost after each chat. It also comes with an admin dashboard where you can set pricing for different models (supports both per-token and per-chat billing) and offers visualizations of usage stats. See the GitHub repo [OpenWebUI Monitor](https://github.com/VariantConst/OpenWebUI-Monitor) for details!

**Reply by MindSetFPS** [reply]
Yes! I actually already installed it yesterday and it was very good, although a bit challenging because i dont speak chinese. I was thinking of asking if you were interesting in adding english and spanish support.

**Reply by VariantConst** [reply]
> Yes! I actually already installed it yesterday and it was very good, although a bit challenging because i dont speak chinese. I was thinking of asking if you were interesting in adding english and spanish support.

It already supports English. Just click the Earth icon at the top of the header. Perhaps I should implement automatic language detection.
<img width="410" alt="image" src="https://github.com/user-attachments/assets/241e7d24-7905-42fe-a590-65747f855b24">

**Reply by pbpippen** [reply]
This is neat. I tried to deploy on my end but couldn't get it up. Used docker compose and tried to access gui on port 7878. I must be doing something wrong!

**Reply by VariantConst** [reply]
> This is neat. I tried to deploy on my end but couldn't get it up. Used docker compose and tried to access gui on port 7878. I must be doing something wrong!

Please open an issue and include detailed debug information.

**Reply by arunbugkiller** [reply]
@VariantConst while my OpenwebUI monitor is up and it is fetching all the models. I am not able to fetch user and usage data. There are no logs as well in the backend that I can refer to.

**Reply by VariantConst** [reply]
> @VariantConst while my OpenwebUI monitor is up and it is fetching all the models. I am not able to fetch user and usage data. There are no logs as well in the backend that I can refer to.当我的 OpenwebUI 监视器启动并正在获取所有模型时，我无法获取用户和使用数据。后端也没有我可以参考的日志。

Probably you configured some environment variables wrongly. See the FAQ. If the problem still exists, please open a issue.

**Reply by arunbugkiller** [reply]
Thanks @VariantConst, It was a problem with enabling openwebui-monitor as a "function" on OpenWeBUI. I hade to enable "Global". 

However, do I need to add Balance now for everyuser ? Because that would be challenging for snew signups. How can i bifurcate that

**Reply by VariantConst** [reply]
> However, do I need to add Balance now for everyuser ? Because that would be challenging for snew signups. How can i bifurcate that

See the environment variable `INIT_BALANCE`.

**Comment by MNBussmann** [comment]
I would also like to see a feature like this. A full front-end metrics display might be a bit much to begin with, maybe a /metrics endpoint that could be read by Prometheus and other similar software would be a good start! This way the data can be extracted and external visualizers like Grafana can be used to display the data.  This would go a long way of making things more transparent for administrators.

For information on how such a metrics endpoint would look like check out the [Prometheus documentation on exposition formats](https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md)

As an example, [Kubernetes uses this format to publish metrics data](https://kubernetes.io/docs/concepts/cluster-administration/system-metrics/), as well as many other projects.

**Comment by rgaricano** [comment]
Another approach, with influxdb and grafana, the script does not take into account the usage of the users, but could be easily implemented:
 https://jorgedelacruz.uk/2025/02/15/looking-for-the-perfect-dashboard-influxdb-telegraf-and-grafana-part-xlvii-monitoring-open-webui/

**Comment by nojaf** [comment]
Hello @tjbck, 

I would be interested in contributing some metrics through OpenTelemetry. I was thinking of starting with something like the total number of logged-in users. Would you be open to such a PR?

**Reply by VariantConst** [reply]
That would be appreciated.
